troubleshooting.md

```diff
@@ -0,0 +1,551 @@
+
+[kni@provisioner]$ sudo virsh vol-delete --vol /var/lib/libvirt/images/kni7-xf6fq-bootstrap 
+Vol /var/lib/libvirt/images/kni7-xf6fq-bootstrap deleted
+
+[kni@provisioner]$ sudo virsh vol-delete --vol /var/lib/libvirt/images/kni7-xf6fq-bootstrap.ign 
+Vol /var/lib/libvirt/images/kni7-xf6fq-bootstrap.ign deleted
+```
+
+Once this is done delete the VM including the storage pool:
+
+```
+[kni@provisioner]$ sudo virsh destroy kni7-xf6fq-bootstrap
+Domain kni7-xf6fq-bootstrap destroyed
+
+[kni@provisioner]$ sudo virsh undefine kni7-xf6fq-bootstrap
+Domain kni7-xf6fq-bootstrap has been undefined
+```
+Or simply make use of 
+
+```
+BOOTSTRAP=$(virsh list --all | grep "$CLUSTER.*bootstrap" | tail -1 | awk '{print $2}')
+virsh destroy $BOOTSTRAP
+virsh undefine $BOOTSTRAP
+virsh vol-delete $BOOTSTRAP default
+virsh vol-delete $BOOTSTRAP.ign default
+```
+
+* You will get the IP of this VM by issuing `sudo virsh net-dhcp-leases` usually always the private IP of this VM is `172.22.0.2` , you can login to this VM using the private SSH key with core as its user:
+
+```
+# ssh -i ~/.ssh/id_rsa core@172.22.0.2
+```
+
+and then issuing the journalctl command to inspect the logs:
+
+```
+[core@localhost ~]$ journalctl -b -f -u bootkube.service
+-- Logs begin at Thu 2019-12-05 10:23:22 UTC. --
+Feb 05 10:29:51 localhost bootkube.sh[3119]: https://etcd-0.kni7.cloud.lab.eng.bos.redhat.com:2379 is unhealthy: failed to commit proposal: context deadline exceeded
+Feb 05 10:29:51 localhost bootkube.sh[3119]: https://etcd-1.kni7.cloud.lab.eng.bos.redhat.com:2379 is unhealthy: failed to commit proposal: context deadline exceeded
+Feb 05 10:29:51 localhost bootkube.sh[3119]: Error: unhealthy cluster
+Feb 05 10:29:51 localhost podman[7743]: 2020-02-05 10:29:51.487999651 +0000 UTC m=+5.434743491 container died a62fd32f5b22e117e1fa5885ae956c5085f3ac5cc47fe88256b8282a9b45695d (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:b3f6e79c60f60a3a8b8422adf82182de632ba2594b34a5b2e56e3564186efe77, name=etcdctl)
+```
+Once logged in to this VM, please check if all the pods(dnsmasq,mariadb,httpd, ironic pods etc) are in running state. 
+
+To check the logs of all the containers , please the below commands:
+
+```
+sudo podman logs ironic-api
+```
+You will get the error messages if the master nodes are not `powered on` by inspecting the ironic pods. 
+
+Once the ironic poewers on the master nodes, verify if RHCOS images are pushed to the cluster, once again inspect the logs of httpd and ironic pods logs from Bootstrap VM. 
+
+## BootstrapVM cannot download RHCOS image
+
+After the Bootstrap VM goes up comes a phase where podman raises 2 containers:
+
+```
+ipa-downloader
+coreos-downloader
+```
+
+These both containers will download resources from (in our case) the jump / bastion host since its a disconnected install.This is the relevant section where is this step performed, modify it in order to let the provisioner to access to the resources:
```

Comment:
(in our case) -- you should generalize this. Maybe talk about what it looks like in a connected vs disconnected state. What do they do?

Reply:
Or better yet, whats the error your trying to solve versus giving information

Reply:
Added a note which states the difference between the two types of scenarios.